regress/unittests/sshkey/common.c

```diff
@@ -83,48 +86,100 @@ load_bignum(const char *name)
 	return ret;
 }
 
-const BIGNUM *
+BIGNUM *
 rsa_n(struct sshkey *k)
 {
-	const BIGNUM *n = NULL;
+	BIGNUM *n = NULL;
+#if (OPENSSL_VERSION_NUMBER < 0x30000000L)
+	RSA *rsa = NULL;
+	BIGNUM *res = NULL;
+#endif
 
 	ASSERT_PTR_NE(k, NULL);
-	ASSERT_PTR_NE(k->rsa, NULL);
-	RSA_get0_key(k->rsa, &n, NULL, NULL);
+	ASSERT_PTR_NE(k->pkey, NULL);
+#if (OPENSSL_VERSION_NUMBER >= 0x30000000L)
+	EVP_PKEY_get_bn_param(k->pkey, OSSL_PKEY_PARAM_RSA_N, &n);
```

Comment:
Does it make sense to have one routine, `ssh_gen_bn_from_rsa` and then move the several clusters of these ifdef/else/endif lines into one place?

Reply:
I am not sure I understand how it can be done. Both named params in new API and key params in old RSA API are not pretty convenient to deal with from an universal function